### PR TITLE
Add support of multiple RRULE

### PR DESCRIPTION
**Pull request by fabien-michel**
When an event got multiple RRULE.
Exemple:
```
RRULE:FREQ=WEEKLY;BYDAY=TH;COUNT=20
RRULE:FREQ=MONTHLY;BYDAY=2MO;COUNT=2
```

It was surprisingly easy to do, and surprising that it hadn't been done before. So I doubt I'm missing anything.

### CODE REVIEW

**Comment by fabien-michel** [comment]
Ok, I think you can review it.
Be really careful, all the tests are passing but I'm not really confident about changes I made. Especially the part modifying `rdate` conditions where I changed single-rrule UNTIL value to the max UNTIL value.

File: recurring_ical_events.py [hunk 1]
```diff
@@ -213,27 +213,32 @@ def __init__(self, component, keep_recurrence_attributes=False):
...
-        else:
-            self.rrule = None
+        self.rule = rruleset(cache=True)
+        _component_rules = component.get("RRULE", None)
+        self.until = None
+        if _component_rules:
+            if not isinstance(_component_rules, list):
+                _component_rules = [_component_rules]
+            else:
+                _dedup_rules=[]
```
Comment (niccokunzmann) [inline recurring_ical_events.py:223]: I will just add a few comments in the process and they might not be necessary for merge. They are partially so I understand the code and you know if I misunderstand something or miss it :)

I think, I would add a comment for the purpose here.
```
# Remove duplicates from rules
```

I wonder if deduplication is necessary but definitely nice.

File: recurring_ical_events.py [hunk 2]
```diff
@@ -213,27 +213,32 @@ def __init__(self, component, keep_recurrence_attributes=False):
...
+                for _rule in _component_rules:
+                    if _rule not in _dedup_rules:
+                        _dedup_rules.append(_rule)
+                _component_rules = _dedup_rules
+    
+            for _rule in _component_rules:
+                rrule = self.create_rule_with_start(_rule.to_ical().decode())
+                self.rule.rrule(rrule)
+                if rrule.until and (not self.until or compare_greater(rrule.until, self.until)):
+                    self.until = rrule.until
```
Comment (niccokunzmann) [inline recurring_ical_events.py:233]: That makes sense... I wonder what the until value is actually used for. Need to look... TODO

**Review by niccokunzmann — APPROVED** [review]
Thanks, this looks really good!

I am pretty sure that this is better than before. Since this is a test-driven library, I am confident that you did not break anything! Thanks for testing, too, because that allows us to be confident!

**Comment by niccokunzmann** [comment]
@fabien-michel Thanks a lot for the nice contribution. Feel free to add a new PR if you find that you would like to change the code a bit more!

**Comment by niccokunzmann** [comment]
I release this under v2.2.1.

**Comment by niccokunzmann** [comment]
Because of the clean work, I give you write access. With this, you can also start to release a new version, review and merge other PRs.

**Pull request merged**